File: src/tauri/commands/watch.ts
import { core } from '@tauri-apps/api'

export function startDirectoryWatch(
  sourceDir: string,
  outputDir: string,
  includeSubfolders: boolean,
) {
  return core.invoke<void>('start_directory_watch', {
    sourceDir,
    outputDir,
    includeSubfolders,
  })
}

export function stopDirectoryWatch() {
  return core.invoke<void>('stop_directory_watch')
}

File: src-tauri/src/tauri_commands/watch.rs
use tauri::AppHandle;

use crate::sys::watch;

#[tauri::command]
pub async fn start_directory_watch(
    app: AppHandle,
    source_dir: String,
    output_dir: String,
    include_subfolders: bool,
) -> Result<(), String> {
    watch::start(&app, &source_dir, &output_dir, include_subfolders)
}

#[tauri::command]
pub async fn stop_directory_watch(app: AppHandle) -> Result<(), String> {
    watch::stop(&app);
    Ok(())
}
